include/zephyr/fs/zms.h

```diff
@@ -82,125 +80,126 @@ struct zms_fs {
  * @param data Pointer to the data to be written
- * @param len Number of bytes to be written (maximum 64 KB)
+ * @param len Number of bytes to be written (maximum 64 KiB)
  *
  * @return Number of bytes written. On success, it will be equal to the number of bytes requested
- * to be written. When a rewrite of the same data already stored is attempted, nothing is written
- * to flash, thus 0 is returned. On error, returns negative value of errno.h defined error codes.
+ * to be written or 0.
+ * When a rewrite of the same data already stored is attempted, nothing is written to flash,
+ * thus 0 is returned. On error, returns negative value of error codes defined in `errno.h`.
  */
 ssize_t zms_write(struct zms_fs *fs, uint32_t id, const void *data, size_t len);
 
 /**
  * @brief Delete an entry from the file system
  *
- * @param fs Pointer to file system
- * @param id Id of the entry to be deleted
+ * @param fs Pointer to the file system.
+ * @param id ID of the entry to be deleted
  * @retval 0 Success
- * @retval -ERRNO errno code if error
+ * @retval -ERRNO Negative errno code on error
  */
 int zms_delete(struct zms_fs *fs, uint32_t id);
 
 /**
  * @brief Read an entry from the file system.
  *
- * @param fs Pointer to file system
- * @param id Id of the entry to be read
+ * @param fs Pointer to the file system.
+ * @param id ID of the entry to be read
  * @param data Pointer to data buffer
- * @param len Number of bytes to be read (or size of the allocated read buffer)
+ * @param len Number of bytes to read at most
  *
  * @return Number of bytes read. On success, it will be equal to the number of bytes requested
- * to be read. When the return value is less than the number of bytes requested to read this
- * indicates that ATE contain less data than requested. On error, returns negative value of
- * errno.h defined error codes.
+ * to be read or less than that if the stored data has a smaller size than the requested one.
+ * On error, returns negative value of error codes defined in `errno.h`.
  */
 ssize_t zms_read(struct zms_fs *fs, uint32_t id, void *data, size_t len);
 
 /**
  * @brief Read a history entry from the file system.
  *
- * @param fs Pointer to file system
- * @param id Id of the entry to be read
+ * @param fs Pointer to the file system.
+ * @param id ID of the entry to be read
  * @param data Pointer to data buffer
  * @param len Number of bytes to be read
  * @param cnt History counter: 0: latest entry, 1: one before latest ...
  *
  * @return Number of bytes read. On success, it will be equal to the number of bytes requested
  * to be read. When the return value is larger than the number of bytes requested to read this
  * indicates not all bytes were read, and more data is available. On error, returns negative
- * value of errno.h defined error codes.
+ * value of error codes defined in `errno.h`.
  */
 ssize_t zms_read_hist(struct zms_fs *fs, uint32_t id, void *data, size_t len, uint32_t cnt);
```

Comment:
Did you make the choice to keep this function public from the get-go? https://github.com/zephyrproject-rtos/zephyr/pull/77930/files#r1795053334

Reply:
Pretty sure this is used in nvs for getting previous entries (assume they have not been written over). Have heard of people using this